nnvm/tests/python/compiler/test_top_level1.py

```diff
@@ -365,6 +365,129 @@ def forward(x):
     inputs = [('x', (1, 3, 28, 28), x)]
     helper(y, inputs, dtype, forward)
 
+def verify_lrn(n, c, h, w, size, axis, bias, alpha, beta):
+    x = sym.Variable("x")
+    y = sym.lrn(x, size=size, axis=axis, bias=bias, alpha=alpha, beta=beta)
+    dtype = "float32"
+    dshape = (n, c, h, w)
+    x_np = np.random.uniform(size=dshape).astype(dtype)
+
+    def lrn_python(a_np, size, axis, bias, alpha, beta):
+        """Local response normalization operator numpy implementation.
+
+        Parameters
+        ----------
+        a_np : numpy.ndarray
+            4-D with shape [batch, in_channel, in_height, in_width]
+
+        size : int
+            normalisation window size
+
+        axis : int
+            input data layout channel axis
+
+        bias : float
+            offset to avoid dividing by 0. constant value
+
+        alpha : float
+            contant value
+
+        beta : float
+            exponent constant value
+
+        Returns
+        -------
+        b_np : np.ndarray
+            4-D with shape [batch, out_channel, out_height, out_width]
+        """
+        axis0, axis1, axis2, axis3 = a_np.shape
+        radius = size // 2
+        sqr_sum = np.zeros(shape=a_np.shape).astype(a_np.dtype)
+        sqr_sum_up = np.zeros(shape=a_np.shape).astype(a_np.dtype)
+        lrn_out = np.zeros(shape=a_np.shape).astype(a_np.dtype)
+        def sum_dot_values(i, j, k, l):
+            axis_size = a_np.shape[axis]
+            if (axis == 1):
+                #NCHW layout
+                sum_start = j-radius if j-radius >= 0 else 0
+                sum_end = j+radius+1 if j+radius+1 < axis_size else axis_size
+                sqr_sum[i, j, k, l] = sum(a_np[i, sum_start:sum_end, k, l] * \
+                                          a_np[i, sum_start:sum_end, k, l])
+            elif (axis == 3):
+                #NHWC layout
+                sum_start = l-radius if l-radius >= 0 else 0
+                sum_end = l+radius+1 if l+radius+1 < axis_size else axis_size
+                sqr_sum[i, j, k, l] = sum(a_np[i, j, k, sum_start:sum_end] * \
+                                          a_np[i, j, k, sum_start:sum_end])
+
+        for i in range(axis0):
```

Comment:
use broadcasting semantics

Reply:
@tqchen , I have tried using broadcast operations further on this changes, but here the sum is doing based on window size and the window move across the axis. could you please elaborate your suggestion

Reply:
OK, never mind, it is fine to keep this one as for loop